core/src/avm2/class.rs

```diff
@@ -1114,3 +1153,19 @@ impl<'gc> Class<'gc> {
         c_link.set(ClassLink::LinkToInstance(i_class));
     }
 }
+
+#[derive(Clone, Copy)]
+pub enum BuiltinType {
+    // `int`, `uint`, `Number`, `Boolean`, `Object`, `String`, `void` builtin
+    // classes
+    Int,
+    Uint,
+    Number,
+    Boolean,
+    Object,
+    String,
+    Void,
```

Comment:
Nit: grouping all primitives types together (by putting `Void` just after `Boolean`) would lead to better codegen for primitive checks.